inc/functions.inc.php

```diff
@@ -44,6 +44,85 @@ function st_register_widget() {
     }
 }
 
+function tp_disp_boolean($bool_text)
+{
+    $bool_text = (string)$bool_text;
+    if (empty($bool_text) || '0' === $bool_text || 'false' === $bool_text) {
+        return 'false';
+    }
+
+    return 'true';
+}
+
+/**
+ * Add custom filter on edit posts page
+ * 
+ * Filter posts by year
+ */
+
+//var_dump(get_option('taxopress_external_taxonomies')['category']['show_filters']);
+
+$show_filters = tp_disp_boolean(get_option('taxopress_external_taxonomies')['category']['show_filters']);
+//var_dump($show_filters);
+
+if ( $show_filters == 'true' ) {
+    add_action('restrict_manage_posts', 'taxopress_add_posts_filter');
+    add_filter( 'parse_query', 'taxopress_filter_by_year');
+} 
+
+
+
+function taxopress_add_posts_filter() {
+
+    global $wpdb, $table_prefix;
+    
+    $post_type = (isset($_GET['post_type'])) ? $_GET['post_type'] : 'post';
+
+    //only add filter to posts
+    if ($post_type == 'post'){
+        
+        //query database to get a list of years for the specific post type:
+        $values = array();
+        $query_years = $wpdb->get_results("SELECT year(post_date) as year from ".$table_prefix."posts
```

Comment:
I don't see the reasons to query database in this requirement. I'll suggest reviewing the issue properly to understand the requirement and do it the right way.

For adding taxonomy dropdown, WordPress already provide an API `wp_dropdown_categories` which is all needed to show dropdown of taxonomies filter